### PR TITLE
CLI build script for a qr verify circuit

### PR DESCRIPTION
**Pull request by vuvoth**
## Motivation

This is CLI build script for qr verify circuit. A snarkjs lib gen chunk zkey, after try figure out(also ref zk-email-verify). 
 We see we need to use a diff snarkjs fork. Check additional context for more details. 
This PR only support build zkey file with backend proofgen (or cli).

## Change Summary

You can check packages/circuits/build.sh for more detail.
We did: 
- Install depends 
- Gen zkey file 
- Proof and verify with valid proof 

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [ ] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] PR has a [change set](https://github.com/privacy-scaling-explorations/anon-aadhaar/CHANGELOG.md)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bug fix, or chore)
- [ ] PR includes documentation if necessary.

## Additional Context

- [Merge chunking in for better memory use on frontends #352](https://github.com/iden3/snarkjs/issues/352)
- [Build script on zkemail](https://github.com/zkemail/zk-email-verify/tree/main/packages/twitter-verifier-circuits/scripts)

### CODE REVIEW

File: packages/circuits/build.sh [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	node ./node_modules/.bin/snarkjs groth16 setup $R1CS_PATH $PTAU_PATH $PARTIAL_ZKEYS_DIR/circuit_0000.zkey
+
+    echo "test random" | NODE_OPTIONS='--max-old-space-size=8192' \
+	node ./node_modules/.bin/snarkjs zkey contribute $PARTIAL_ZKEYS_DIR/circuit_0000.zkey $PARTIAL_ZKEYS_DIR/circuit_final.zkey --name="1st Contributor Name" -v 
+
+    yarn remove snarkjs
+    yarn add snarkjs@0.7.0
+    yarn
+
+    NODE_OPTIONS=--max-old-space-size=8192 \
```
Comment (Meyanis95) [inline packages/circuits/build.sh:73]: Why do we need to run the same command 2 times with different versions of `snarkjs`?

File: packages/circuits/build.sh [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    if [ ! -d $PARTIAL_ZKEYS_DIR ]; then
+        mkdir -p $PARTIAL_ZKEYS_DIR
+    fi
+
+    echo "TRUSTED SETUP FOR DEVELOPMENT - PLEASE, DON'T USE IT IN PRODUCT!!!!"
+
+    circom ./circuits/qr_verify.circom  --r1cs --wasm -o $BUILD_DIR -l ./node_modules
+
+    yarn add snarkjs@git+https://github.com/vb7401/snarkjs.git#24981febe8826b6ab76ae4d76cf7f9142919d2b8
```
[the remark this answers is not in the document]
Comment (vuvoth) [reply packages/circuits/build.sh]: Yes, You're right! I removed those line!

File: packages/circuits/build.sh (hunk 1 above)
Comment (vuvoth) [reply packages/circuits/build.sh:66]: I removed those lines!

File: packages/circuits/build.sh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	node ./node_modules/.bin/snarkjs groth16 setup $R1CS_PATH $PTAU_PATH $PARTIAL_ZKEYS_DIR/circuit_0000.zkey
+
+    echo "test random" | NODE_OPTIONS='--max-old-space-size=8192' \
+	node ./node_modules/.bin/snarkjs zkey contribute $PARTIAL_ZKEYS_DIR/circuit_0000.zkey $PARTIAL_ZKEYS_DIR/circuit_final.zkey --name="1st Contributor Name" -v 
+
+    NODE_OPTIONS=--max-old-space-size=8192 \
+	node ./node_modules/.bin/snarkjs groth16 setup $R1CS_PATH $PTAU_PATH $PARTIAL_ZKEYS_DIR/circuit_0000.zkey
+
+    echo "test random" | NODE_OPTIONS='--max-old-space-size=8192' \
+	node ./node_modules/.bin/snarkjs zkey contribute $PARTIAL_ZKEYS_DIR/circuit_0000.zkey $PARTIAL_ZKEYS_DIR/circuit_final_nonchunk.zkey --name="1st Contributor Name" -v
```
Comment (Meyanis95) [inline packages/circuits/build.sh:70]: I think we should change it to "2nd Contributor Name"

**Comment by Meyanis95** [comment]
LGTM!
Just added one last comment + need to fix the conflict

**Comment by saleel** [comment]
I think the JS based script in Email Wallet make more sense for us (we only need one circuit) - https://github.com/zkemail/email-wallet/blob/main/packages/circuits/scripts/dev-setup.ts

@vuvoth  Also do we really need chunked zkeys? What is the size of zkey with QR verification and extraction?

**Review by Meyanis95 — APPROVED** [review]
LGTM

**Pull request merged**